userdata_mining/mining/miner.py

```diff
@@ -1050,20 +1046,32 @@ def mine_data(self):
                 embedding.embed(x) for x in hangouts_data]
             self.messages_embeddings = [
                 x for x in self.messages_embeddings if x is not None]
-            
+
             # Cache embeddings
             with open(f'{self.data_path}/saved/embeddings/hangouts.pickle', 'wb') as f:
                 pickle.dump(self.messages_embeddings, f)
         else:
             self.messages_embeddings = []
-        
+
         if chats_data:
             info('Embedding Google Chat data. This may take a while.')
             self.chats_embeddings = [
                 embedding.embed(x) for x in chats_data]
         else:
             self.chats_embeddings = []
 
+        if mail_data:
+            info('Embedding email data. This may take a while.')
+            self.email_embeddings = [embedding.embed(x) for x in mail_data]
+            self.email_embeddings = [
+                x for x in self.email_embeddings if x is not None]
+
+            # Cache email embeddings
+            with open(f'{self.data_path}/saved/embeddings/mail.pickle', 'wb') as f:
+                pickle.dump(self.email_embeddings, f)
+        else:
+            self.email_embeddings = []
+
```

Comment:
@yrahul3910 please check me on this. I'm not sure why mail embedding was a different format than everything else, please lmk if this screws something up

Reply:
This seems right to me. I'm not sure what you mean it was a different format. In any case, as long as the embeddings are pickled, it should be okay.

Reply:
Previously, the mail data embedding was in an `else` block instead of an if block and was separated from the other embeddings, down past the `if data is None` checks